### PR TITLE
tokio: fix API doc examples

### PR DESCRIPTION
**Pull request by carllerche**

### CODE REVIEW

File: tokio/src/runtime/threadpool/mod.rs [hunk 1]
```diff
@@ -183,9 +183,7 @@ impl Runtime {
 
     /// Signals the runtime to shutdown once it becomes idle.
     ///
-    /// Returns a future that completes once the shutdown operation has
-    /// completed.
-    ///
+    /// Blocks the current thread until the sutdown operation has completed.
```
Comment (LucioFranco) [inline tokio/src/runtime/threadpool/mod.rs]:
```suggestion
    /// Blocks the current thread until the shutdown operation has completed.
```

File: tokio/src/runtime/threadpool/mod.rs [hunk 2]
```diff
@@ -208,25 +206,24 @@ impl Runtime {
...
+        let inner = self.inner.take().unwrap();
+        e.block_on(inner.pool.shutdown_on_idle());
     }
 
     /// Signals the runtime to shutdown immediately.
     ///
-    /// Returns a future that completes once the shutdown operation has
-    /// completed.
-    ///
+    /// Blocks the current thread until the sutdown operation has completed.
```
Comment (LucioFranco) [inline tokio/src/runtime/threadpool/mod.rs]:
```suggestion
    /// Blocks the current thread until the shutdown operation has completed.
```

File: tokio/src/timer.rs [hunk 3]
```diff
@@ -30,45 +30,44 @@
...
-//!     Delay::new(when)
-//!         .map_err(|e| panic!("timer failed; err={:?}", e))
-//!         .and_then(|_| {
-//!             println!("Hello world!");
-//!             Ok(())
-//!         })
-//! })
+//! #[tokio::main]
+//! async fn main() {
+//!     let when = tokio::clock::now() + Duration::from_millis(100);
```
Comment (LucioFranco) [inline tokio/src/timer.rs]: why the change from `Instant::now`?

Comment (carllerche) [reply tokio/src/timer.rs]: We should move away from `Instant::now()` in general. `clock::now()` can be hooked into from tests.

Comment (LucioFranco) [reply tokio/src/timer.rs]: Just to follow up then we should document that all the timer modules? and maybe even provide api's in delay to auto call out to tokio::clock?
